### ISSUE TITLE
<br>tag at the end of <pre>-tag gives error

### ISSUE THREAD

**Issue by MrMamen** [prettier/prettier#6028]
**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeADgJzgPgM4QC2cMAFgJZQDmqARpjqgPRa4gA0IE6M50eyUAENMmCAHcACiIQCUQgG4RyAEw4h6QsAGsSAZXRbKVZDEwBXOJ1IxCAGwDqFeHkNg4e2eV4LvAT2RwPAFOSjw4TBhJTCEqQiFkADMhO3DOACs8AA8AIRidfSFiABlKOCSUtJBMrL1jOzgARXMIeArUqxBDTHDMQJt7dSxKGAdVMmQADgAGTiwIcIcY9EDWXoVyzmwAR3NybGjY+Paq8MJyUwtOvHqmlrakZI7OGCFaMZUJpAAmF5jyOzGADCRGOKCg0E2IHM4QAKm85E9wgBfZFAA)
```sh
--parser html
```

**Input:**
```html
<pre>something<br></pre>
```

**Output:**
```html
printChild@https://prettier.io/lib/standalone.js:20894:291
printChildren/<@https://prettier.io/lib/standalone.js:20881:102
map@https://prettier.io/lib/standalone.js:15766:21
printChildren@https://prettier.io/lib/standalone.js:20825:24
genericPrint$2@https://prettier.io/lib/standalone.js:20709:39
callPluginPrintFunction@https://prettier.io/lib/standalone.js:15912:18
printGenerically/res<@https://prettier.io/lib/standalone.js:15863:16
printComments@https://prettier.io/lib/standalone.js:15586:17
printGenerically@https://prettier.io/lib/standalone.js:15862:22
printAstToDoc@https://prettier.io/lib/standalone.js:15874:16
coreFormat@https://prettier.io/lib/standalone.js:16193:16
format@https://prettier.io/lib/standalone.js:16441:73
formatWithCursor@https://prettier.io/lib/standalone.js:16457:12
withPlugins/<@https://prettier.io/lib/standalone.js:30711:15
format@https://prettier.io/lib/standalone.js:30719:12
formatCode@https://prettier.io/worker.js:234:21
handleMessage@https://prettier.io/worker.js:185:18
self.onmessage@https://prettier.io/worker.js:155:14

```

**Comment by Compynerd255** [comment]
I'm getting this issue as well - the error I'm seeing is

    SyntaxError: Unexpected closing tag ":pre". It may happen when the tag has already been closed by another tag. For more info see https://www.w3.org/TR/html5/syntax.html#closing-elements-that-have-implied-end-tags ([line]:[col])

This occurs with any closing tag - it appears that `<br>` is being read by the parser as an opening tag without a closing tag, even though it's part of the set of tags that get implicitly closed. I suspect that the auto-closing logic is (incorrectly) assumed to close the previous tag that the object is currently in, even if it isn't a non-auto-closing tag. In addition, this bug also occurs with other parsers, including Babel.

As an aside, this bug makes it impossible to use Prettier with [DOMPurify](https://github.com/cure53/DOMPurify), because DOMPurify generates this style of `<br>` tags and cannot be configured to do otherwise.

**Comment by fisker** [comment]
This was fixed with a wrong solution in `v2.0.0`, reverted #8645

**Comment by fisker** [comment]
This one already have PR https://github.com/prettier/prettier/pull/8841, I think we can fix this in 2.1
